packages/@lwc/synthetic-shadow/src/faux-shadow/focus.ts

```diff
@@ -302,9 +302,22 @@ function skipHostHandler(event: FocusEvent) {
     }
 }
 
+// Async because the current implementation relies on the capture phase of the focusin event and we
+// don't want to enable the focus navigation logic until the capture phase is complete.
+let asyncEnableFocusNavigationTimer: number | undefined;
+function asyncEnableFocusNavigation() {
+    if (!isUndefined(asyncEnableFocusNavigationTimer)) {
+        return;
+    }
+    asyncEnableFocusNavigationTimer = setTimeout(() => {
```

Comment:
I can't think of any escenario in which this will break, but I'm sure there is something out there. I have a couple of suggestions:

* use a micro-task instead of a macro-task, to reduce the possibility to some user-code doing the wrong thing, like simulating clicks.
* look at potential solutions to determine when the event switches from capture to bubble, I believe it is possible, but might require adding listeners at multiple levels. E.g.: if during capture, you add a bubble counter part that expires quickly, but knows the event instance, that means you can know, when observing this event, when to synchronously debounce the flag.

Reply:
> use a micro-task instead of a macro-task, to reduce the possibility to some user-code doing the wrong thing, like simulating clicks.

Unfortunately, we can't use microtasks here because those execute immediately after each `focusin`.

> look at potential solutions to determine when the event switches from capture to bubble, I believe it is possible, but might require adding listeners at multiple levels. E.g.: if during capture, you add a bubble counter part that expires quickly, but knows the event instance, that means you can know, when observing this event, when to synchronously debounce the flag.

I guess we might be able to avoid async if we did that, but it would also open us up to a fragile situation where our polyfill stops working if user code did something like `event.stopPropagation()`.